packages/flutter/test/widget/animated_size_test.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+void main() {
+  testWidgets('AnimatedSize test', (WidgetTester tester) async {
```

Comment:
You should test what happens when the parent gives the AnimatedSize tight constraints and then varies those constraints over time.  (There shouldn't be any animation running in that case---i.e., no transient callbacks.)